ApollosProject/apollos-prototype
# Adds Apollos Config

## DESCRIPTION

### What does this PR do, or why is it needed?
Adds an ApollosConfig package that will automatically load and manage all the configuration variables for the apollos project and all dependencies, once imported and called. 

This package can be used by any of the server side applications that need access to ENV variables. As of right now, we will keep the documentation for variables inside the `config.yml` file, but in the future we should probably annotate the `config.yml` file more like a piece of documentation itself and keep our values out of `git`.  

Also includes a bunch of changes to support testing.

### What design trade-offs/decisions were made
As of right now, we are keeping the `config.yml` inside version control. In the future, this should be removed from version control and replaced with a more lightweight `config.example.yml`

### How do I test this PR?
1. Clone.
2. `yarn && yarn start` and click around the app to make sure everything works.

### What automated tests did you write?
n/a

## TODO:

- [x] Closes #351 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
